sdk/ml/azure-ai-ml/azure/ai/ml/exceptions.py

```diff
@@ -247,17 +243,15 @@ def __init__(
         message: str,
         no_personal_data_message: str,
         *args,
-        target: ErrorTarget = ErrorTarget.UNKNOWN,
-        error_category: ErrorCategory = ErrorCategory.UNKNOWN,
+        target: Union[str, ErrorTarget] = ErrorTarget.UNKNOWN,
+        error_category: Union[str, ErrorTarget] = ErrorCategory.UNKNOWN,
```

Comment:
categories should match

Reply:
same above

Reply:
Same comment as above, let's fix these all over. Once the bug above is resolved, we should be good to make `error_category: Union[str, ErrorCategory]` everywhere.

Reply:
fixed